src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -132,12 +133,10 @@ private Task OnProjectFactoryCompletedAsync()
         {
             var previousProjectContext = _currentAggregateProjectContext;
             var newProjectContext = await UpdateProjectContextAsync().ConfigureAwait(false);
+
             if (previousProjectContext != newProjectContext)
             {
-                // Dispose existing subscriptions.
-                DisposeAndClearSubscriptions();
```

Comment:
Thanks for checking. I think regardless, NuGet restorer should be fine handling duplicate updates as you are just kicking off restore when the update happens, you are not building any state based on design time builds.